src/objects/tests/test_merge_patch.py

```diff
@@ -0,0 +1,29 @@
+from unittest import TestCase
+
+from objects.api.utils import merge_patch
+
+
+class MergePatchTests(TestCase):
+    def test_merge_patch(self):
+
+        test_data = [
+            ({"a": "b"}, {"a": "c"}, {"a": "c"}),
+            ({"a": "b"}, {"b": "c"}, {"a": "b", "b": "c"}),
+            ({"a": "b"}, {"a": None}, {}),
+            ({"a": "b", "b": "c"}, {"a": None}, {"b": "c"}),
+            ({"a": ["b"]}, {"a": "c"}, {"a": "c"}),
+            ({"a": "c"}, {"a": ["b"]}, {"a": ["b"]}),
+            ({"a": {"b": "c"}}, {"a": {"b": "d", "c": None}}, {"a": {"b": "d"}}),
+            ({"a": [{"b": "c"}]}, {"a": [1]}, {"a": [1]}),
+            (["a", "b"], ["c", "d"], ["c", "d"]),
+            ({"a": "b"}, ["c"], ["c"]),
+            ({"a": "foo"}, None, None),
+            ({"a": "foo"}, "bar", "bar"),
+            ({"e": None}, {"a": 1}, {"e": None, "a": 1}),
+            ([1, 2], {"a": "b", "c": None}, {"a": "b"}),
+            ({}, {"a": {"bb": {"ccc": None}}}, {"a": {"bb": {}}}),
```

Comment:
Please add a test with the same `target` and `patch`to check idempotency something like
```{"a": "b"}, {"a": "b"}, {"a": "b"}),```